ion-schema/src/violation.rs

```diff
@@ -61,13 +61,11 @@ impl Violation {
     pub fn flattened_violations(&self) -> Vec<&Violation> {
         let mut flattened_violations = Vec::new();
         self.flatten_violations(&mut flattened_violations);
+        flattened_violations.sort_by(|a, b| (a.constraint).cmp(&b.constraint));
         flattened_violations
     }
 
     fn flatten_violations<'a>(&'a self, flattened: &mut Vec<&'a Violation>) {
-        if self.violations.is_empty() {
-            flattened.push(self);
-        }
```

Comment:
I'm not sure if this is correct. If there is any violation, and you flatten them, you should always get a non-empty list of violations, right?

Reply:
That seems right. I guess I was thinking more about returning child violations for the current violation. More like a flattened version of `violations()` method which returns a violation tree.

Reply:
Changed this behavior back to return the current violation as well. `flattened_violations` will always return atleast 1 violation.